rfcs/text/0017_url_service.md

```diff
@@ -0,0 +1,462 @@
+   */
+  create(
+    urlGeneratorId: string,
+    state: unknown,
+    slug?: string
+  ): Promise<{ slug: string; url: URL }>;
+
+  /**
+   * Delete a short URL.
+   * 
+   * @param slug The slug (ID) of the short URL.
+   * @return Returns true if deletion was successful.
+   */
+  delete(slug: string): Promise<boolean>;
+
+  /**
+   * Fetch short URL.
+   * 
+   * @param slug The slug (ID) of the short URL.
+   */
+  get(slug: string): Promise<{ slug: string; shortUrl: object; url: URL }>;
+
+  /**
+   * Same as `get()` but it also increments the "view" counter and the
+   * "last view" timestamp of this short URL.
+   * 
+   * @param slug The slug (ID) of the short URL.
+   */
+  resolve(slug: string): Promise<{ slug: string; shortUrl: object; url: URL }>;
+}
+```
+
+Note, that in the new service to generate a short URL the developer will have to
+use a URL generator.
+
+```ts
+const shortUrl = await share.url.shortUrls.create({
+  generator: 'DISCOVER_DEEP_LINKS',
+  state: {
+    indexPattern: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx',
+  },
+});
+```
+
+These short URL will be stored in saved objects of type `url` and will be
+automatically migrated once a URL generator becomes deprecated.
+
+
+## HTTP endpoints
+
+
+### Short URL CRUD+ HTTP endpoints
+
+Below HTTP endpoints are designed to work specifically with short URLs:
+
+| HTTP method           | Path                                | Description                                                                                                                              |
+|-----------------------|-------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| __POST__              | `/api/short_url`                    | Endpoint for creating new short URLs.                                                                                                    |
+| __GET__               | `/api/short_url/<slug>`             | Endpoint for retrieving information about an existing short URL.                                                                         |
+| __DELETE__            | `/api/short_url/<slug>`             | Endpoint for deleting an existing short URL.                                                                                             |
+| __POST__              | `/api/short_url/<slug>`             | Endpoint for updating information about an existing short URL.                                                                           |
+| __GET__ or __POST__   | `/api/short_url/<slug>/_resolve`    | Same as `GET /api/short_url/<slug>`, but this endpoint also increments the short URL access count counter and the last access timestamp. |
+| __GET__ or __POST__   | `/api/short_url/<slug>/_redirect`   | Redirects the user using HTTP 301 (permanent) redirect to the destination long URL.                                                      |
```

Comment:
im confused by this GET or POST endpoints, could you split this up to make it clearer ?

Reply:
I've cleaned it up and updated the RFC, see the latest text.